modules/src/ics04_channel/msgs/chan_open_try.rs

```diff
@@ -0,0 +1,322 @@
+#![allow(clippy::too_many_arguments)]
+
+use crate::ics03_connection::connection::validate_version;
+use crate::ics04_channel::channel::{ChannelEnd, Counterparty, Order};
+use crate::ics04_channel::error::{Error, Kind};
+use crate::ics23_commitment::commitment::CommitmentProof;
+use crate::ics24_host::identifier::{ChannelId, ConnectionId, PortId};
+use crate::{proofs::Proofs, tx_msg::Msg, Height};
+
+use tendermint::account::Id as AccountId;
+
+use serde_derive::{Deserialize, Serialize};
+use std::str::FromStr;
+
+/// Message type for the `MsgChannelOpenTry` message.
+const TYPE_MSG_CHANNEL_OPEN_TRY: &str = "channel_open_try";
+
+///
+/// Message definition for the second step in the channel open handshake (`ChanOpenTry` datagram).
+///
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct MsgChannelOpenTry {
+    port_id: PortId,
+    channel_id: ChannelId,
+    channel: ChannelEnd,
+    counterparty_version: String,
+    proofs: Proofs,
+    signer: AccountId,
+}
+
+impl MsgChannelOpenTry {
+    pub fn new(
+        port_id: String,
+        channel_id: String,
+        channel_version: String,
+        order: i32,
+        connection_hops: Vec<String>,
+        counterparty_port_id: String,
+        counterparty_channel_id: String,
+        counterparty_version: String,
+        proof_init: CommitmentProof,
+        proofs_height: Height,
+        signer: AccountId,
+    ) -> Result<MsgChannelOpenTry, Error> {
+        let connection_hops: Result<Vec<_>, _> = connection_hops
+            .into_iter()
+            .map(|s| ConnectionId::from_str(s.as_str()))
+            .collect();
+
+        let version =
```

Comment:
Note to self: this was changed in a separate PR: https://github.com/informalsystems/ibc-rs/pull/388/files#diff-a9dd86861adb60ba50e159c43913d6f20a5d406115a2ef4d17c7d0725054c7cc